### PR TITLE
Compatibility fixes for PowerDNS 4.x

### PR DESCRIPTION
**Pull request by nickmarden**

### CODE REVIEW

File: recipes/_source.rb [hunk 1]
```diff
@@ -36,7 +36,7 @@
 pdns_source_dir = pdns_dir(pdns_filename)
 
 remote_file pdns_filepath do
-  source lazy { node['pdns'][flavor]['source']['url'] }
+  source node['pdns'][flavor]['source']['url']
```
Comment (martinisoft) [inline recipes/_source.rb]: This is a pretty old workaround. Are you still using Chef 10 or a version earlier than 11.14?

Comment (nickmarden) [inline recipes/_source.rb]: I'm using Chef 11.10, which is the most recent Chef 11.x that I can get on AWS OpsWorks :-/

**Comment by alexbacchin** [comment]
will this ever be merged?

**Comment by nickmarden** [comment]
Yes @martinisoft I was wondering the same thing?

**Comment by martinisoft** [comment]
Let me review it today with @onlyhavecans and we can see about getting it reviewed and merged. Sorry for the delay 👍

File: libraries/helpers.rb [hunk 2]
```diff
@@ -58,7 +60,7 @@ def pdns_package_module_requirements
 
 def pdns_source_module_requirements
   modules = node['pdns']['authoritative']['backends']
-  required_packages = []
+  required_packages = %w(autoconf automake bison flex g++ git libboost-all-dev libtool make pkg-config ragel)
```
Comment (martinisoft) [inline libraries/helpers.rb]: Wouldn't a lot of these packages come from build-essential?

File: recipes/_source.rb [hunk 3]
```diff
@@ -66,11 +67,13 @@
   modules = "--with-modules='#{node['pdns']['authoritative']['backends'].join(' ')}' "
   binary_string = 'pdns_server'
 
-  execute 'pdns: bootstrap' do
-    # This insanity is documented in the README
-    command './bootstrap && ./bootstrap'
-    cwd pdns_source_dir
-    not_if "/usr/local/sbin/#{binary_string} --version 2>&1 | grep #{version}"
+  if major_version < 4
```
Comment (martinisoft) [inline recipes/_source.rb]: This can be coupled with the `not_if` as an `only_if { major_version > 4 }`

File: recipes/_source.rb (hunk 1 above)
Comment (martinisoft) [inline recipes/_source.rb]: That makes sense. Thank you for clarifying 👍

File: recipes/_source.rb (hunk 3 above)
Comment (nickmarden) [inline recipes/_source.rb]: OK. I think I looked at that possibility briefly and was unclear how `not_if` and `only_if` work together.

File: libraries/helpers.rb (hunk 2 above)
Comment (nickmarden) [inline libraries/helpers.rb]: Probably, but I don't think `build-essential` is listed as a requirement anywhere so I went with the minimal set of packages that were needed for this particular build.

Comment (martinisoft) [inline libraries/helpers.rb]: I'd just go ahead and make build-essential a cookbook dependency then for now.

**Comment by martinisoft** [comment]
Once the comments are corrected we can look at a merge. Thanks again for sending us this pull request 👍

**Pull request closed without merging**